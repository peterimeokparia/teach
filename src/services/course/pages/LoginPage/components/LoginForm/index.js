import { 
useState } from 'react';

const LoginForm = ({ 
error, 
loading, 
handleLoginUser }) => {
const [email, setEmail] = useState('');
const [password, setPassword] = useState('');

const handleSubmit = (e) => {
  e.preventDefault();
};

return   (    
  <div className="LoginPage"> 
      <form onSubmit={ e => handleSubmit(e)}>
        <label>  
            Email
          <input
            data-cy={`email`}
            id="email"
            name="email"
            type="email"
            // value={email}
            onChange={ e => setEmail( e.target.value ) }
            placeholder="email"
            autoComplete="username"
          >
          </input>
        </label>
        <label>
          Password   
          <input
            data-cy={`password`}
            id="password"
            name="password"
            type="password"
            // value={password}
            onChange={ e => setPassword( e.target.value ) }
            placeholder="password"
            autoComplete="current-password"
          >
          </input>
        </label>
          <div>                   
          </div>
          { error  && (<div className="error"> { error.message }</div>)}
          <div></div>
          <button
            data-cy={`submit-login`}
            id="submit-login"
            type="submit"
            disabled={loading}
            onClick={e => handleLoginUser(email, password)}
          >
              Sign In
          </button>
        </form>    
  </div> 
  );
};

export default LoginForm;