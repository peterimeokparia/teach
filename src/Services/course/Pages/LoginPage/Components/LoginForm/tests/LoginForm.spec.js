import React from 'react';
import LoginForm  from 'services/course/pages/LoginPage/components/LoginForm';
import renderer from 'react-test-renderer';

describe('LoginForm', () =>  {  

  beforeEach(() => {
    jest.spyOn(Date.prototype, 'toLocaleTimeString')
        .mockImplementation(() => '12:00:00');
    jest.spyOn(Date.prototype, 'toLocaleDateString')
      .mockImplementation(() => 'AM 1/01/2021');
  });

  it('renders as expected', async () => {
    const tree = renderer.create(
        <LoginForm /> 
    );
      expect(tree.toJSON()).toMatchSnapshot();
  })
});




