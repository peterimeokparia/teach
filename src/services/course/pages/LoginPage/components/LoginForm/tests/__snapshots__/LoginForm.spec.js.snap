// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders as expected 1`] = `
<div
  className="LoginPage"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Email
      <input
        autoComplete="username"
        data-cy="email"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
      />
    </label>
    <label>
      Password
      <input
        autoComplete="current-password"
        data-cy="password"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
      />
    </label>
    <div />
    <div />
    <button
      data-cy="submit-login"
      id="submit-login"
      onClick={[Function]}
      type="submit"
    >
      Sign In
    </button>
  </form>
</div>
`;
